### PR TITLE
Added subprotocol specification and Conn as a parameter in events

### PR DESCRIPTION
**Pull request by brettjbush**
Hi rgamba,

I really like your event driven websocket implementation for Go! Best and easiest to use library I've found for the language.

I have a few things I would like to add to your library:

1. The ability to specify a subprotocol on Dial
2. Pass the main Conn object as an argument to events in order to allow OnConnected and OnMessage to run Send(msg)

If you would like to me to make any adjustments so as to fit in more with your implementation, I would be happy to make them. Thanks for a great library!

Brett

### CODE REVIEW

File: evtwebsocket.go [hunk 1]
```diff
@@ -104,15 +104,15 @@ func (c *Conn) onMsg(msg []byte) {
...
+	go c.OnMessage(msg, c)
 }
 
 func (c *Conn) close() {
 	c.ws.Close()
 	c.closed = true
 	if c.Reconnect {
 		for {
-			if err := c.Dial(c.url); err == nil {
+			if err := c.Dial(c.url, ""); err == nil {
```
Comment (rgamba) [inline evtwebsocket.go]: It seems we would like to save the subprotocol as a parameter of Conn so we can pass it here on reconnect

**Review by rgamba — CHANGES_REQUESTED** [review]
Nice! I posted a minor detail regarding the subprotocol argument, it should be saved in the Conn struct so we can pass it on reconnect. Also it would be awesome if you could update the examples and readme as your changes effectively break them.
Thanks

**Comment by brettjbush** [comment]
Thanks for feedback,

Sure thing, I'll make the changes you've described.

File: evtwebsocket.go (hunk 1 above)
Comment (brettjbush) [reply evtwebsocket.go]: Just added this change and README/example updates in the latest commit. If there is anything more I can do, feel free to let me know. Thanks!

**Review by rgamba — COMMENTED** [review]
Looks good to me. Thanks!

**Pull request merged**